src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathAbstractNodeTest.java

```diff
@@ -0,0 +1,375 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.xpath;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class XpathAbstractNodeTest {
+
+    private static AbstractNode abstractNode;
+
+    @Before
+    public void init() {
+        abstractNode = Mockito.mock(AbstractNode.class, Mockito.CALLS_REAL_METHODS);
```

Comment:
@romani 
We do the same for [AbstractCheck](https://github.com/MEZk/checkstyle/blob/0093555b4a1487ee48d0b49dd9a59c53d9c730da/src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractCheckTest.java#L39-L39), [AbstractViolationReporter](https://github.com/MEZk/checkstyle/blob/256237f2c5fb17da4e926832310609e285155697/src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractViolationReporterTest.java#L38-L38), [AbstractFileSetCheck](https://github.com/MEZk/checkstyle/blob/e9bea8c67bd2c629624d33f58611d4bc562c0233/src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java#L38-L38), [AbstractLoader](https://github.com/MEZk/checkstyle/blob/e9bea8c67bd2c629624d33f58611d4bc562c0233/src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractLoaderTest.java#L38-L38) to satisfy coverage rate. 

If it should be tested though the concreate implementation, then we need to move the unsupported methods to the concrete classes and copy-paste them to all subclasses.
See https://github.com/checkstyle/checkstyle/pull/4815#discussion_r130214936